src/os_sensor_nodelet.cpp

```diff
@@ -160,16 +160,22 @@ class OusterSensor : public OusterClientBase {
         int lidar_port =
             config.udp_port_lidar ? config.udp_port_lidar.value() : 0;
         int imu_port = config.udp_port_imu ? config.udp_port_imu.value() : 0;
+        auto udp_dest = config.udp_dest ? config.udp_dest.value() : "";
 
         std::shared_ptr<sensor::client> cli;
-        if (lidar_port != 0 && imu_port != 0) {
+        if (config.mtp_dest) {
+            auto mtp_dest = config.mtp_dest.value();
+            // use the full init_client to recieve data via multicast 
+            cli = sensor::init_client(hostname, udp_dest, mtp_dest,
+                                      sensor::MODE_UNSPEC, sensor::TIME_FROM_UNSPEC,
+                                      lidar_port, imu_port);
+        } else if (lidar_port != 0 && imu_port != 0) {
```

Comment:
Per this [comment](https://github.com/ouster-lidar/ouster_example/pull/480#discussion_r1103453915), we can't modify **sensor_config**, this we need to parse mt_dest and save it to a data member of **OusterSensor** nodelet.

```suggestion
        if (!this->mtp_dest.empty()) {
            // use the full init_client to recieve data via multicast 
            cli = sensor::init_client(hostname, udp_dest, this->mtp_dest,
                                      sensor::MODE_UNSPEC, sensor::TIME_FROM_UNSPEC,
                                      lidar_port, imu_port);
        } else if (lidar_port != 0 && imu_port != 0) {
```

Reply:
Resolved in [commit](https://github.com/ouster-lidar/ouster-ros/pull/57/commits/49e131c0d5250159fd933a703e352edaf186003f)